File: Day 3/Task 1/Task 1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include <algorithm>

int main()
{
	std::ifstream fhandle("day_3_1.txt");
	std::string content;
	int prioritySum = 0;

	while (std::getline(fhandle, content)) {
		std::size_t half_length = content.length() / 2;
		auto it = std::find_first_of(content.begin(), content.begin() + half_length, content.begin() + half_length, content.end());
		
		if (std::isupper(*it))
			prioritySum += *it - 'A' + 27;
		else
			prioritySum += *it - 'a' + 1;


	}
	
	std::cout << "The sum of the priorites of items that appear in both compartments of the rucksack is: " << prioritySum;



	


}

File: Day 3/Task 1/Task_1.py
def ItemPriority(item):
    if (item.isupper()):
        return ord(item) - ord('A') + 27

    return ord(item) - ord('a') + 1


fhandle = open("day_3_1.txt")

prioritySum = 0

for line in fhandle:
    line = line.rstrip()
    half_length = len(line) // 2

    for i in range(half_length):
        commonItemIndex = line.find(line[i], half_length, len(line))
        if commonItemIndex != -1:
            prioritySum += ItemPriority(line[commonItemIndex])
            break

print(prioritySum)

